src/main/java/org/truffleruby/core/module/ModuleNodes.java

```diff
@@ -2584,4 +2584,13 @@ protected Object doClass(RubyClass rubyClass) {
             return rubyClass.isSingleton;
         }
     }
+
+    @Primitive(name = "refined_class")
```

Comment:
The Primitive name should reflect in which module/class it's defined, so it should be `refinement_refined_class`.
Ideally it'd be a CoreMethod though, for that we need to add `RefinementNodes`, which I think makes sense.
We should then move the `refinement_import_methods` Primitive there.